### PR TITLE
feat: 995 - support of the new "product type" filter for "get product"

### PR DESCRIPTION
**Pull request by monsieurtanuki**
### What
- Now we have a "product type" filter for "get product"

### Fixes bug(s)
- Closes: #995

### Files
New file:
* `product_type_filter.dart`: Filter on product type for API get product queries.

Impacted files
* `api_not_food_get_product_test.dart`: now testing only in v3 with all product type filters
* `old_product_result.dart`: deprecated class `OldProductResult`
* `open_food_api_client.dart`: deprecated method `getOldProduct`
* `openfoodfacts.dart`: export new file `product_type_filter.dart`
* `product_query_configurations.dart`: "product_type_filter" as new optional "get product" parameter; minor refactoring

### CODE REVIEW

File: lib/src/model/product_type_filter.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import 'product_type.dart';
+
+/// Filter on product type for API get product queries.
+class ProductTypeFilter implements OffTagged {
+  const ProductTypeFilter._(this.offTag);
+
+  ProductTypeFilter(final ProductType productType)
+      : offTag = productType.offTag;
+
+  static const all = ProductTypeFilter._('all');
```
Comment (g123k) [inline lib/src/model/product_type_filter.dart:11]: I think you missed the type here

**Review by g123k — COMMENTED** [review]
Just one minor comment

File: lib/src/model/product_type_filter.dart (hunk 1 above)
Comment (monsieurtanuki) [reply lib/src/model/product_type_filter.dart:11]: Indeed. Strange the lint check didn't send a warning.
I'll fix that.

**Review by g123k — APPROVED** [review]
LGTM 👌
Thank you!

**Comment by monsieurtanuki** [comment]
Thank you @g123k for your review!

**Pull request merged**